osu.Game/Screens/Play/HUD/HitErrorDisplay.cs

```diff
@@ -0,0 +1,209 @@
+                    },
+                    new Container
+                    {
+                        Anchor = reversed ? Anchor.CentreRight : Anchor.CentreLeft,
+                        Origin = reversed ? Anchor.CentreRight : Anchor.CentreLeft,
+                        AutoSizeAxes = Axes.X,
+                        RelativeSizeAxes = Axes.Y,
+                        Child = arrow = new SpriteIcon
+                        {
+                            Anchor = Anchor.Centre,
+                            Origin = Anchor.Centre,
+                            RelativePositionAxes = Axes.Y,
+                            Icon = reversed ? FontAwesome.Solid.ChevronRight : FontAwesome.Solid.ChevronLeft,
+                            Size = new Vector2(8),
+                        }
+                    },
+                }
+            });
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(OsuConfigManager config)
+        {
+            config.BindWith(OsuSetting.ScoreMeter, type);
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            HitWindows.SetDifficulty(beatmap.Value.BeatmapInfo.BaseDifficulty.OverallDifficulty);
+
+            bar.AddRange(new Drawable[]
+            {
+                new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = ColourInfo.GradientVertical(colours.Yellow.Opacity(0), colours.Yellow),
+                    Height = (float)((getMehHitWindows() - HitWindows.Good) / (getMehHitWindows() * 2))
+                },
+                new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = colours.Green,
+                    Height = (float)((HitWindows.Good - HitWindows.Great) / (getMehHitWindows() * 2))
+                },
+                new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = colours.BlueLight,
+                    Height = (float)(HitWindows.Great / getMehHitWindows())
+                },
+                new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = colours.Green,
+                    Height = (float)((HitWindows.Good - HitWindows.Great) / (getMehHitWindows() * 2))
+                },
+                new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = ColourInfo.GradientVertical(colours.Yellow, colours.Yellow.Opacity(0)),
+                    Height = (float)((getMehHitWindows() - HitWindows.Good) / (getMehHitWindows() * 2))
+                }
+            });
+
+            type.BindValueChanged(onTypeChanged, true);
+        }
+
+        private double getMehHitWindows()
+        {
+            // In case if ruleset has no Meh hit windows (like Taiko)
+            if (HitWindows.Meh == 0)
+                return HitWindows.Good + 40;
```

Comment:
why are we doing this? does stable do this?

Reply:
In stable (in taiko) Meh hit windows presented as well even if you can't hit them

Reply:
That sounds like a stable limitation. Let's not do that.